main.nf

```diff
@@ -56,29 +56,28 @@ log.info ""
 log.info "Input: $params.root"
 root = file(params.root)
 /* Watch out, files are ordered alphabetically in channel */
-    in_data = Channel
-        .fromFilePairs("$root/**/{*tracking.trk,*anat.nii.gz}",
-                        size: 2,
-                        maxDepth:2,
-                        flat: true) {it.parent.name}
-
-    atlas_anat = Channel.fromPath("$params.atlas_anat")
-    atlas_config = Channel.fromPath("$params.atlas_config")
-    atlas_directory = Channel.fromPath("$params.atlas_directory")
-
-    if (params.atlas_centroids) {
-        atlas_centroids = Channel.fromPath("$params.atlas_centroids/*.trk")
-    }
-    else {
-        atlas_centroids = Channel.empty()
-    }
-
-(anat_for_registration, anat_for_reference, tractogram_for_recognition) = in_data
-    .map{sid, anat, tractogram -> 
-        [tuple(sid, anat),
-        tuple(sid, anat),
-        tuple(sid, tractogram)]}
-    .separate(3)
+Channel
+    .fromPath("$root/**/*tracking*.trk",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .set{tractogram_for_recognition}
+
+Channel
+    .fromPath("$root/**/*fa.nii.gz",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .into{anat_for_registration;anat_for_reference_c;anat_for_reference_b}
+
+atlas_anat = Channel.fromPath("$params.atlas_anat")
```

Comment:
Could be nice to test if all the mandatory arguments are differents than false. Something like in TractoFlow just to be more robust.